cartesi-rollups/challenges/overview.md

```diff
@@ -0,0 +1,84 @@
+
+### Solution strategy
+
+Following community-built applications, it's clear that "higher-level development frameworks" might greatly benefit usability. The community could explore the development of these frameworks, allowing developers to abstract away several concepts, benefit from code completion, reuse standard code such as wallet implementations, and much more. Engaging the community in these areas could lead to some innovative solutions.
+
+## Design for upgradability
+
+### Challenge
+
+The current **v0.9** release incorporates mechanisms that account for the evolution of Cartesi Rollups and the DApps built on it, including the ability to modify smart contract behaviors related to validator operations, dispute strategies, and other aspects.
+The previous use of the diamonds design pattern has been replaced, allowing for a simpler approach to upgrade and modify smart contracts. This simplification enables developers to "point" specific functions of the Rollups framework to a new smart contract, which can implement the desired behavior differently. For example, to upgrade the consensus mechanism from "Authority" to "Quorum", developers can call the [migrateToConsensus](../api/json-rpc/sol-output.md/#migratetoconsensus) function and pass the address of the smart contract that implements the Quorum behavior. While this new approach is intended to make the process more convenient for developers, it is clear that there might still be room for improvement to further streamline the experience for day-to-day developers and the general audience.
+In particular, upgrading the back-end code of a DApp is an important concern that may arise due to the discovery of a bug in the application itself or the software stack that it utilizes. This requires the DApp's Cartesi Machine to be safely updated, for which there is currently no standard mechanism or guideline. As a reminder, individual DApps can of course always choose to simply not be upgradable at all.
+
+### Solution strategy
+
+Following the **v1.0** mainnet release, an interesting area for research could be to explore methods for supporting an upgrade procedure for DApps that are inclined to allow it. A key concern in this research domain is finding a way, or a set of best practices, to securely handle asset ownership continuity – for instance, how to transfer assets safely from an older version of an application to its newer incarnation. Given that an updated DApp would entail utilizing a different Cartesi Machine, the challenge is to devise a robust method to migrate the previous machine's state, inclusive of user assets, into this new framework.
```

Comment:
'for instance, how to maintain asset ownership of the DApp after a version upgrade'